core/evm.go

```diff
@@ -63,14 +63,17 @@ func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common
 	}
 
 	// Set goldTokenAddress
-	caller := &CeloBackend{config, statedb.Copy()}
-	registry, err := abigen.NewRegistryCaller(contracts_config.RegistrySmartContractAddress, caller)
-	if err != nil {
-		log.Error("Failed to access registry!", "err", err)
-	}
-	goldTokenAddress, err := registry.GetAddressForOrDie(&bind.CallOpts{}, contracts_config.GoldTokenRegistryId)
-	if err != nil {
-		log.Error("Failed to get address for GoldToken!", "err", err)
+	var goldTokenAddress common.Address
+	if config.IsCel2(header.Time) {
```

Comment:
does this imply that everytime you use the value you will also check the flag?

Reply:
This happens once per block and is then stored in the BlockContext. The precompile only checks `BlockContext.GoldTokenAddress`.

Reply:
We might even be able to get rid of this altogether, since we have fixed the `GoldTokenAddress` to the current mainnet address and plan to use that across all Cel2 networks.